src/main/java/com/example/controller/SimulationController.java

```diff
@@ -90,13 +91,28 @@ public void hide() {
 
     private void setDefaultSettings() {
         algorithmSettings.getSettings().put("depth",
-                new AlgorithmSetting<>("depth", 1, Integer.class, (value) -> value >= 0));
+                new AlgorithmSetting<>("depth", 1, Integer.class, (value) -> value > 0));
         algorithmSettings.getSettings().put("phase",
-                new AlgorithmSetting<>("phase", 1, Integer.class, (value) -> value >= 0));
+                new AlgorithmSetting<>("phase", 1, Integer.class, (value) -> value > 0));
         algorithmSettings.getSettings().put("q",
                 new AlgorithmSetting<>("q", 1, Integer.class, (value) -> value >= 0));
         algorithmSettings.getSettings().put("time",
-                new AlgorithmSetting<>("time", 1, Integer.class, (value) -> value >= 0));
+                new AlgorithmSetting<>("time", 1, Integer.class, (value) -> value > 0));
+    }
+
+    public void setSettingsValidation(GraphController graphController) {
+        if(graphController.getGraph().numVertices() == 0) {
+            algorithmSettings.getSettings().get("depth").setValidateArgument((value) ->  (Integer)value == 0);
+            algorithmSettings.getSettings().get("q").setValidateArgument((value) -> (Integer)value == 0);
+        }
```

Comment:
For empty graph any arguments should be invalid. Try this way:
(value) ->  false

Reply:
Fixed